pkg/performance/collector.go

```diff
@@ -0,0 +1,281 @@
+package performance
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+)
+
+// PointCollector performs one-shot data collection
+type PointCollector interface {
+	Type() MetricType
+	Name() string
+
+	// Collect performs a single collection and returns the metrics
+	Collect(ctx context.Context) (interface{}, error)
+
+	Capabilities() CollectorCapabilities
+}
+
+// ContinuousCollector performs ongoing data collection with streaming output
+type ContinuousCollector interface {
+	Type() MetricType
+	Name() string
+
+	// Start begins continuous collection and returns a channel for streaming results
+	Start(ctx context.Context) (<-chan interface{}, error)
+
+	// Stop halts continuous collection and cleans up resources
+	Stop() error
+
+	Status() CollectorStatus
+	LastError() error
+	Capabilities() CollectorCapabilities
+}
+
+type CollectorCapabilities struct {
+	SupportsOneShot    bool
+	SupportsContinuous bool
+	RequiresRoot       bool
+	RequiresEBPF       bool
+	MinKernelVersion   string
+}
+
+type BasePointCollector struct {
```

Comment:
if we had the general Collector interface described in the above comment, then this could be renamed to `BaseCollector`

Because Collect is not implemented, the naming is confusing.